swagger-ui/hooks.go

```diff
@@ -0,0 +1,378 @@
+	// the test address - TODO IMPORTANT make sure this lines up with startchain.sh
+	address = "kava1ffv7nhd3z6sych2qpqkk03ec6hzkmufy0r2s4c"
+	return address
+}
+
+func getKeynameAndPassword() (keyname string, password string) {
+	keyname = "vlad"      // TODO - IMPORTANT this must match the keys in the startchain.sh script
+	password = "password" // TODO - IMPORTANT this must match the keys in the startchain.sh script
+	return keyname, password
+}
+
+// this should send a delegation
+func sendDelegation() {
+	// get the address
+	address := getTestAddress()
+	// get the keyname and password
+	keyname, password := getKeynameAndPassword()
+
+	addrFrom, err := sdk.AccAddressFromBech32(address) // validator
+	if err != nil {
+		panic(err)
+	}
+
+	// helper methods for transactions
+	cdc := app.MakeCodec() // make codec for the app
+
+	// get the keybase
+	keybase := getKeybase()
+
+	// get the validator address for delegation
+	valAddr, err := sdk.ValAddressFromBech32("kavavaloper1ffv7nhd3z6sych2qpqkk03ec6hzkmufyz4scd0") // **FAUCET**
+	if err != nil {
+		panic(err)
+	}
+
+	// create delegation amount
+	delAmount := sdk.NewInt64Coin(sdk.DefaultBondDenom, 1000000)
+	delegation := staking.NewMsgDelegate(addrFrom, valAddr, delAmount)
+	delegationToSend := []sdk.Msg{delegation}
+
+	// send the delegation to the blockchain
+	sendMsgToBlockchain(cdc, address, keyname, password, delegationToSend, keybase)
+}
+
+// this should send a MsgUndelegate
+func sendUndelegation() {
+	// get the address
+	address := getTestAddress()
+	// get the keyname and password
+	keyname, password := getKeynameAndPassword()
+
+	addrFrom, err := sdk.AccAddressFromBech32(address) // validator
+	if err != nil {
+		panic(err)
+	}
+
+	// helper methods for transactions
+	cdc := app.MakeCodec() // make codec for the app
+
+	// get the keybase
+	keybase := getKeybase()
+
+	// get the validator address for delegation
+	valAddr, err := sdk.ValAddressFromBech32("kavavaloper1ffv7nhd3z6sych2qpqkk03ec6hzkmufyz4scd0") // **FAUCET**
+	if err != nil {
+		panic(err)
+	}
+
+	// create delegation amount
+	undelAmount := sdk.NewInt64Coin(sdk.DefaultBondDenom, 1000000)
+	undelegation := staking.NewMsgUndelegate(addrFrom, valAddr, undelAmount)
+	delegationToSend := []sdk.Msg{undelegation}
+
+	// send the delegation to the blockchain
+	sendMsgToBlockchain(cdc, address, keyname, password, delegationToSend, keybase)
+
+}
+
+func getKeybase() crkeys.Keybase {
+	// create a keybase
+	// TODO - IMPORTANT - this needs to be set manually and does NOT work with tilde i.e. ~/ does NOT work
+	// TODO - QUESTION - should we read the path from a configuration file?
+	home, err := os.UserHomeDir()
+	if err != nil {
+		panic(err)
+	}
+	keybase, err := keys.NewKeyBaseFromDir(home + "/kavatmp/")
+	if err != nil {
+		panic(err)
+	}
+
+	return keybase
+}
+
+// sendMsgToBlockchain sends a message to the blockchain via the rest api
+func sendMsgToBlockchain(cdc *amino.Codec, address string, keyname string,
```

Comment:
I think this method should return the tx hash, and then for transactions that need to be included we should include a `awaitTxConfirmed()` method which does something like 
```
confrirmed = false
for !confirmed {
  resp, err := http.Get("https://localhost:1317/txs/{hash}")
  // sleep for 5 seconds if tx not found and poll again
 // when tx is found, set confirmed to true and return
}
```

I think this will solve the issue of having to run the tests twice.